tests/ui/test_ui_tools.py

```diff
@@ -2755,3 +2755,45 @@ def test_update_widget(self, mocker, caption, expected_cursor_position,
             caption, cursor_position=expected_cursor_position
         )
         assert isinstance(mocked_button._w, AttrMap)
+
+    @pytest.mark.parametrize('link, handle_narrow_called', [
+        (SERVER_URL + '/#narrow/stream/1-Stream-1', True),
+        (SERVER_URL + '/user_uploads/some/path/image.png', False),
+        ('https://foo.com', False),
```

Comment:
`(SERVER_URL + '/#narrow/stream/asdasdgasdgasdg', ?),` could be in a message typed in manually, I think? What should happen?